packages/data-broker/src/tests/services/weather/DateSanitiser.test.js

```diff
@@ -0,0 +1,123 @@
+import { expect } from 'chai';
+import { DateSanitiser } from '../../../services/weather/DateSanitiser';
+import { mockNow, resetMocks } from './testutil';
+
+describe('DateSanitiser', () => {
+  beforeEach(() => {
+    mockNow(1549360800 * 1000); // (2019-02-05 10:00 UTC)
+  });
+
+  afterEach(() => {
+    resetMocks();
+  });
+
+  it('throws when start > end', () => {
+    const functionCall = () => new DateSanitiser().sanitise('2019-01-20', '2019-01-10');
+    expect(functionCall).to.throw('Start date must be before (or equal to) end date');
+  });
+
+  describe('limits on earliest data', () => {
+    /*
+     * This is a limitation of the WeatherBit plan we use, if we request further back it falls over
+     */
+
+    it('both start and end dates are before the earliest available data', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2017-02-20',
+      );
+
+      expect(actualStartDate).to.equal('2018-02-05'); // earliest date
+      expect(actualEndDate).to.equal('2018-02-06'); // earliest + 1 day
+    });
```

Comment:
I think this case should return no data. If you've requested dates that all fall outside of the supported range, you get no data - that seems fine. On the other hand, getting data from a single date that you didn't choose seems like unexpected behaviour that would be hard to predict without knowing the internals

Reply:
Agreed, updated